### PR TITLE
Remove Copy NuGet Package to Azure Blob Store build step

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Remove Copy NuGet Package to Azure Blob Store build step. They are not needed and they don't work.

It will be nice to have it in the 1.9 release branch, but it is not necessary. This PR is just to remove a warning. 

**Motivation and Context**
- Why is this change required? What problem does it solve?

To remove a warning from our packaging pipeline.

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml [hunk 1]
```diff
@@ -562,30 +562,6 @@ jobs:
         python.exe $(Build.SourcesDirectory)\tools\ci_build\github\windows\post_binary_sizes_to_dashboard.py --commit_hash=$(Build.SourceVersion) --size_data_file=binary_size_data.txt --build_project=Lotus --build_id=$(Build.BuildId)
       workingDirectory: '$(Build.BinariesDirectory)'
 
-  - task: AzureFileCopy@3
-    displayName: 'Copy Signed Native NuGet Package to Blob Store'
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:566]: The build step requires AzureRM powershell module, but that module is obsoleted and I don't want to install it on our build machines.

Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:566]: do we have support for the AzureCLI step?
https://docs.microsoft.com/en-us/cli/azure/storage/blob?view=azure-cli-latest#az_storage_blob_upload

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:566]: Yes. But nobody is consuming packages from there. We have a nuget feed for daily packages. If anyone need the nuget package, they would prefer to fetch the packages from the feed.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:566]: And even the feed is not often used.

**Review by jingyanwangms — APPROVED** [review]
We should remove these steps if they don't work and gives warning.

**Pull request merged**